### DISCUSSION TITLE
How do we request access to "buyer_email" via Etsy API after February 5th - MANDATORY REQUIREMENT FOR SHIPPING

### DISCUSSION THREAD

**Discussion by DevBrent** [etsy/open-api#1185] [Q&A]
As you may know, email is a mandatory field for shipping with many providers. We at the same time have been pressured to eliminate the use of dummy email addresses to satisfy this requirement.

A. Does Etsy plan on providing a forwarding email address much like Amazon does?
B. Why does your team offer no mention of HOW to gain permission to continue receiving buyer_email in the announcement: https://github.com/etsy/open-api/discussions/1184 or the emailed announcement. It just says `If your application is not authorized to access this field`
C. How can we request permission to retain buyer_email and phone number for the purposes of fulfillment of the order.

This decision is going to prevent your sellers from shipping their orders after February 5th.

Thanks.

**Comment by sowsinsk** [comment]
I agree, it is **critical** that we understand **if** and **how** our app is able to receive authorization for the buyer_email field.

**Reply by showtroylove** [reply]
They do briefly mention the permission / scope required here at [API Etsy](https://developers.etsy.com/documentation/#commercial-access).

In pertinent part:

Applications with commercial access that use the transaction_r permission scope must request access to the buyer_email field separately. Etsy approves these requests on a case by case basis.

**Reply by Archieversam** [reply]
Are you looking for a way to track and convert majority of your traffic into potential buyers in an easy way? Kindly contact us now for more details or use the link below. THANKS.
https://cutt.ly/Sw0vSP70

**Comment by jeremylaiyt** [comment]
Yes, I agree as well.
I would like to understand what is the suggested approach from Etsy moving forward as well, and how can we obtain approval to access this field in the future. The buyer's email information is essential for processing orders in our workflow too.

**Comment by mattkingshott** [comment]
Similar agreement here. This looks like another classic example of the Etsy dev team making an arbitrary decision with zero consultation among the dev community. 

Being able to get buyer Email is foundational. Without it, many apps will simply stop working!

**Reply by mattkingshott** [reply]
Also, one month is nowhere near enough lead time for a change this major.

**Comment by annedirkse** [comment]
+1 to this. Without API access to messenger (which we would frankly prefer) we have no practical means of communicating with the buyer about their order. All of our orders require customer approval of proofs, and email is the only way we currently have of sending these in an organized, timely manner and getting a response. 

Unfortunately with Etsy messenger the process is error prone and the response rate from customers is very low. We have brought the API access issue up over the years.

We built our app specifically to be able to manage hundreds of (genuinely handmade) orders at once. I doubt that Etsy is planning to open up API access to messenger, so what is the solution? How do we apply for access? Without access to email our app will need major updates and 1 month is very little time to do that. This will hurt our company and more importantly our customers who are familiar with the process and regular, repeat buyers. 

Please let us know how to apply for access to the email field ASAP.

**Comment by mathieulesniak** [comment]
A big +1 to this. Our french shipping providers all requires at least recipient phone number or recipient email address. Without that, impossible to create shipping slips

**Comment by emergingdzns** [comment]
Just be aware that Etsy staff generally doesn't monitor the discussions. If you need to request having access, you should contact them via developer@etsy.com.

**Reply by annedirkse** [reply]
But of course in the email about this, they say: "Replies to this email will not be monitored, but if you need further assistance, visit our API Developers hub: https://developers.etsy.com/documentation/get-help"
I suspect that none of us are going to get any solution from Etsy.

**Comment by emergingdzns** [comment]
Also, as I understand it, not ALL apps will lose this ability. You should be notified if your app is affected. I am guessing (very wild guess) that apps that are shipping apps won't be affected. Could be wrong though.

**Reply by mattkingshott** [reply]
Arbitrary decision making is not acceptable in a professional environment. It’s either all apps have it or none. Preferential treatment is wrong.

**Reply by alejandroLoz** [reply]
This is a message from one of the lucky "chosen" 3rd party providers:
"We're pleased to inform you that we have a strong relationship with Etsy. Fortunately, we've obtained an exemption from the recent change. This means XXX sellers won't be affected, and we'll still be able to access the buyer's email address from Etsy orders."

**Reply by mattkingshott** [reply]
Which provider was this?

**Reply by jamiethompson** [reply]
We are a shipping app. We filled out the [exemption form](https://docs.google.com/forms/d/1h_lVsqNysY5KeDbgHMKxQI-S6lUu7dBl00gRZoHc5u0/viewform?edit_requested=true) and provided all of the information required. They have responded to say that we will not be exempted.

**Reply by petermeng72** [reply]
same here, bunch of clowns. We used to offer tracked postage with email updates, no longer. Making the customer service on their own website worse for customers and sellers alike. No reasons given as well why they will not apply the exemption to our account too. It is so poor.

**Comment by DriperPulse** [comment]
+1. Without the buyers e-mail address (or replacing that access with the ability to send chat messages via API), the entire workflow of our app fails.

**Reply by annedirkse** [reply]
Even with sending messages by messenger (if that were supported) we had a major problem with that early on because we require a response and tons of customers will reply to the no-reply email notification and then get upset when you don't do what they requested in the response which Etsy bounces and we never see. It's a nightmare scenario and we have... 1 month... to work around it.

**Comment by sowsinsk** [comment]
On the main Open API v3 [documentation page](https://developers.etsy.com/documentation/), under _Commercial Access_, the following is stated:

> Applications with commercial access that use the transaction_r permission scope must request access to the buyer_email field separately. Etsy approves these requests on a case by case basis.

Nothing is mentioned regarding Personal (non-commercial) Apps. Hopefully more information is yet to come.

**Reply by mattkingshott** [reply]
Is this a new addition? I don’t remember having to do this before.

**Reply by sowsinsk** [reply]
According to the Wayback Machine, that sentence has been on the page for a while. I only noticed it while hunting for info on `buyer_email`. Hopefully we'll get a clarification on the process for requesting access, and which apps are eligible.

**Reply by mattkingshott** [reply]
Do you remember having to make a separate request for that? I don’t remember having to do it. 

Maybe if you had an app on v2 of the API, you got it automatically?

**Reply by sowsinsk** [reply]
I do not recall, however my app is Personal, not commercial.

**Reply by annedirkse** [reply]
I had this option selected when I looked at the "Edit Your App" screen this morning, but the option to request it seems to have been removed.

**Reply by mattkingshott** [reply]
I do wonder what life in Etsy dev team must be like…

Happy Nee Year! Let’s disable this? Sure! Sounds good! Should we consult with dev community? Nah, it’ll be fine.

**Reply by annedirkse** [reply]
I just got a message from developer@etsy.com that there is an exemption form that we can fill out - I suggest contacting them. Fingers crossed that it works.

**Reply by mattkingshott** [reply]
Thanks for feedback. 

Did they send you the form? Any more details would be appreciated if you have them. Thanks!

**Reply by annedirkse** [reply]
Yep - the form they sent is here: https://docs.google.com/forms/d/1h_lVsqNysY5KeDbgHMKxQI-S6lUu7dBl00gRZoHc5u0/viewform?edit_requested=true

**Reply by mattkingshott** [reply]
Awesome. Thanks for sharing 👏🏻

**Reply by scottjlaw** [reply]
Be good if they literally had "for providing to shipping providers" as an answer, as "To communicate with the buyer during the fulfillment process" is broad.

How is the buyer email address intended to be used (check all that apply)

- To complete a purchase transaction
- To communicate with the buyer during the fulfillment process
- To follow up with a buyer post-purchase
- For future communication with the buyer
- Other:

**Reply by scottjlaw** [reply]
I got approved for a private app by following that form

**Reply by jamiethompson** [reply]
We got denied after filling out that form. We are a shipping app and we use the buyer email address only to pass on the the shipping provider. I explained this is great detail on the form. We were denied.

**Comment by laloutre87** [comment]
I would like to point out that it's not only the shipping companies that require email nowadays, it's also the mandatory online declaration for Customs. I don't know how it works in the u.s. but here in france, the postal service manual interface for Customs declaration takes about 15min to fill, if you're fast. So if i can't do it using automation, i just realistically can't ship outside of the E.U. anymore  
Tagging @nateswart as he is the one that did the (locked for comments) announcement.

**Reply by mattkingshott** [reply]
Excellent point. There’s a lot of scenarios which the Etsy dev team don’t appear to have considered. That’s why there should have been a consultation period IMO.

**Reply by jamiethompson** [reply]
I explained this to them when I filled out [the form.](https://docs.google.com/forms/d/1h_lVsqNysY5KeDbgHMKxQI-S6lUu7dBl00gRZoHc5u0/viewform?edit_requested=true) They still denied our request for an exemption.

**Comment by scottjlaw** [comment]
+1 shouting into the void of Etsy not listening... will give that form a try though

**Comment by thesignal** [comment]
+1 Same problem here! We need the email to communicate with the buyers about their personalized items - since there is no api for the message system. 

Also we need to pass the email-adress to our shipping provider so the customers get access to multiple actions for their package (allow deposit at their door, change the address, and so on).
This will be a nightmare without the emailaddress :(

**Comment by uzairfiaz-gelato** [comment]
This API endpoint mentions that we will need `email_r` but is not clear why I believe to get the email related to the user:
https://developer.etsy.com/documentation/reference/#operation/getUser

**Comment by sowsinsk** [comment]
Does anyone have any clarification from Etsy on the permitted vs non-permitted use cases for using the buyer's email?

**Reply by mathieulesniak** [reply]
Just received answers to form submissions. They were all rejected with the same answer. 
Anyone else received feedback after the forem submission ?

**Reply by laloutre87** [reply]
Yup. Rejected because "the seller can still access the buyer email in the shop manager". So basically protecting the buyer's email isn't a concern, seller can still collect it as he wants, but making automation of time consuming tasks like Customs declarations, shipping labels etc ...  seems to cross some line.  Un-understandable. Every other marketplaces I know just provide relay emails like orderid@mails.marketplace.com  and relay emails to customer, thus protecting buyer's email without making seller's life worse.

**Reply by mathieulesniak** [reply]
Exactly the same. What a pity that we couldn't have an relay email address as you described.

**Reply by Arnaud-DT** [reply]
Same here. Didn't even bother to provide a reason...

**Reply by scottjlaw** [reply]
I got approved, although it's for a private internal app, not a public one

**Reply by mathieulesniak** [reply]
Unfortunately, even for an internal app, result was the same for me. 
May I ask what is your use case ?

**Reply by scottjlaw** [reply]
Just shipping and customs purposes...

**Reply by thesignal** [reply]
I got also approved for an internal app (which uses it for our previews we are sending and also the shipping provider).

**Reply by mathieulesniak** [reply]
Really don't understand on which criteria Etsy decides if an app should be approved or not.
Internal app, shipping like you, and no luck

**Reply by thesignal** [reply]
Maybe it depends on the amount of orders per month/year?

**Comment by namlv94** [comment]
Etsy has announced: "Any developers impacted by this change will receive an email from Etsy later today." Since we haven't received any notification, does it mean that our app's access to buyer_email will not be restricted?

**Comment by jamiethompson** [comment]
> Yup. Rejected because "the seller can still access the buyer email in the shop manager". So basically protecting the buyer's email isn't a concern, seller can still collect it as he wants, but making automation of time consuming tasks like Customs declarations, shipping labels etc ... seems to cross some line. Un-understandable. Every other marketplaces I know just provide relay emails like [orderid@mails.marketplace.com](mailto:orderid@mails.marketplace.com) and relay emails to customer, thus protecting buyer's email without making seller's life worse.

We got the same response @laloutre87. So they're happy with a seller copying and pasting the buyer's email address into our app (how else can it now work?) rather than just allowing us to import it? If they actually want to protect the buyer's email address then they should implement a simple email relay system like ebay and amazon do. If the seller still has access to the buyer email addressses then they're still not protected at all.

**Comment by Gareth064** [comment]
I've said it on other discussions, but Etsy don't want sellers using third-party apps. They want everyone using their Apps all the time. That's how they get metrics on how their apps are used. I can't see this API every getting any new features, or even the bugs fixes that we reported here.

Look at their latest "release" fixing the documentation. Wow. Amazing work these guys are spending their time on.

They probably dont have a dedicated API team anymore and it's just beat endeavours.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by yohaas** [comment]
Looks like the [form](https://docs.google.com/forms/u/0/d/1h_lVsqNysY5KeDbgHMKxQI-S6lUu7dBl00gRZoHc5u0/closedform) is closed - anyone have any other info on exemptions? Like others, we pass the email along to carriers - some require an email, so it's either that or a fake email address.

**Reply by jminardi** [reply]
Did you ever figure out a new way to ask for exemptions? I have a personal (non-commercial) app that relies on accessing the buyer email to send a proof of their design so I can get approval before beginning the manufacturing. I need this working and it seems like people earlier in the thread were able to get approvals for that.

**Reply by jamiethompson** [reply]
@jminardi I asked for an exemption. For a well-established third party shipping application. We were denied an exemption. I asked for a reconsideration, especially based on our demonstrable relationships with eBay and Amazon, which include annual audits by Deloitte, but no. They just said no again. No explanation. Just no.

**Reply by yohaas** [reply]
I emailed Etsy developer support but have not gotten any meaningful response. For now we're using a fake email address if the buyer email isn't available.

**Comment by turboturtleD** [comment]
For anyone who is using their app internally and also uses Shipstation, I assume that the Etsy-->Shipstation integration will still be provided with the buyer Etsy email addresses.  Knowing that, you could grab the email address from the Shipstation API if it was critical for you to have.  Not a solution of course, but potentially a relatively painless workaround.

**Comment by bindalf** [comment]
hi guys,
any news on getting approved for buyer_email field?

they took it from my live application with no explanation, just that they decided not to give the rights to my app. And no answer to the question what is needed to get the rights.

I also created a new application, the approval for it is pending since over a month.

**Comment by mihnsen** [comment]
Hi guys,
do you have any news on getting approve for `buyer_email`? anyway to ask for exemptions?
We register as a new application and need to send an email to ask for customization of design

**Reply by swiftmyday** [reply]
Not only they removed email, but also shipping address from api. They will remove more and more feature year after year. This should be called Etsy closed-api

**Comment by bindalf** [comment]
no luck, also with a new app. They don't approve
